### PR TITLE
feat: Add StatefulSet rollout tracker

### PR DESCRIPTION
**Pull request by nightkr**
# Description

Extracted from https://github.com/stackabletech/hdfs-operator/pull/571, as requested by https://github.com/stackabletech/hdfs-operator/pull/571#discussion_r1706580600.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [x] Changes are OpenShift compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

File: crates/stackable-operator/src/rollout.rs
Comment (sbernauer) [inline crates/stackable-operator/src/rollout.rs:1]: I think I would prefer to not put more stuff into top-level modules. WDYT of putting it below `status` or `utils`?

Comment (nightkr) [reply crates/stackable-operator/src/rollout.rs:1]: The whole crate is a big `utils` module, honestly. Status I could see.

Comment (nightkr) [reply crates/stackable-operator/src/rollout.rs:1]: 583afb2f9eec14c06a0e7de8d85aac9bdf111ab2

Comment (sbernauer) [reply crates/stackable-operator/src/rollout.rs:1]: :+1:

**Review by sbernauer — APPROVED** [review]
Thanks! FYI: I updated the PR title, hope that's ok

**Comment by nightkr** [comment]
Conventional commits are gonna be the death of us all.

**Pull request merged**